Sources/ParseSwift/Types/ParseError.swift

```diff
@@ -406,3 +406,90 @@ extension ParseError: CustomStringConvertible {
         debugDescription
     }
 }
+
+// MARK: Compare Errors
+extension Error {
+    /**
+     Validates if the given ParseError Code is equal to the error.
+     
+    **Example use case:**
+    ~~~
+    if error.containedInParseError(.objectNotFound)  {
+        //Do stuff
+    }
+    ~~~
+     - parameter errorCode: The ParseError code you want to compare to
+     */
+    public func equalsToParseError(_ errorCode: ParseError.Code) -> Bool {
+        guard let parseError = self as? ParseError else {
+            return false
+        }
+        return parseError.code == errorCode
+    }
+
+    /**
+     Returns an optional ParseError if the given ParseError Code is equal to the error.
+     
+    **Example use case:**
+    ~~~
+    if let parseError = error.equalsToParseError(.objectNotFound)  {
+        print(parseError.description)
+    }
+    ~~~
+     - parameter errorCode: The ParseError code you want to compare to
+     
+     - returns: An optional ParseError
+     */
+    public func equalsToParseError(_ errorCode: ParseError.Code) -> ParseError? {
+        guard let parseError = self as? ParseError else {
+            return nil
+        }
+        if parseError.code == errorCode {
+            return parseError
+        } else {
+            return nil
+        }
+    }
+
+    /**
+     Validates if the given ParseError Codes contains the error.
+     
+    **Example use case:**
+    ~~~
+    if error.containedInParseError([.objectNotFound, .invalidQuery])  {
+        //Do stuff
+    }
+    ~~~
+     - parameter errorCode: The ParseError codes as **Array** you want to compare to
+     */
+    public func containedInParseError(_ errorCode: [ParseError.Code]) -> Bool {
+        guard let parseError = self as? ParseError else {
+            return false
+        }
+        return errorCode.contains(parseError.code)
+    }
+
+    /**
+     Returns an optional ParseError if the given ParseError Codes contains the error.
+     
+    **Example use case:**
+    ~~~
+    if let parseError = error.containedInParseError([.objectNotFound, .invalidQuery])  {
+        print(parseError.description)
+    }
+    ~~~
+     - parameter errorCode: The ParseError codes as **Array** you want to compare to
+     
+     - returns: An optional ParseError
+     */
+    public func containedInParseError(_ errorCode: [ParseError.Code]) -> ParseError? {
+        guard let parseError = self as? ParseError else {
```

Comment:
It's good to match the original method you are methods you are trying to replicate. For example, this should be called `contains`. Ending a method name with `ParseError` is usually something in objective c. 

You can also reduce the lines of this method along with similar ones. I recommend:

```swift
public func contains(_ errorCode: [ParseError.Code]) -> ParseError? {
        guard let error = self as? ParseError,
              errorCode.contains(error.code) == true else {
            return nil
        }
        return error
    }
```

Reply:
Wouldn't this be better? If not, let me know.
Cause it checks if the error is contained in the Array, and it seems to me that `contains` means that it checks if it contains both Errors.

```
public func containedIn(_ errorCode: [ParseError.Code]) -> ParseError? {
        guard let error = self as? ParseError,
              errorCode.contains(error.code) == true else {
            return nil
        }
        return error
    }
```

Reply:
`containedIn` is okay I guess. I was looking at the method you are using for an array, `contains`, but that only takes one element. `containedIn` is okay because you are passing multiple.